torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1052,7 +1052,8 @@ RegisterOperators reg(
            const auto type = node->output()->type()->expect<ClassType>();
            const size_t numAttrs = type->numAttributes();
            return [type, numAttrs](Stack& stack) {
-             auto userObj = c10::ivalue::Object::create(type, numAttrs);
+             auto userObj =
+                 c10::ivalue::Object::create(type->compilation_unit(), type, numAttrs);
```

Comment:
`type->compilation_unit()` <- in the future this won't be able to store an owning pointer to the compilation unit. Should we make CompilationUnit into an intrusive_ptr? Otherwise we would need an shared_from_this() here.

Reply:
I make it a weak_ptr later on, which I think is fine (in particular, we don't want the weak intrusive ptr to accidentally keep the CU alive)